### PR TITLE
Add Tomcat metrics that are tracked via JMX

### PR DESCRIPTION
**Pull request by checketts**
This adds in several additional Tomcat metrics that are tracked/exposed via JMX.

Cache Metrics (counters)
```
tomcat_cache_hit_total 0.0
tomcat_cache_access_total 0.0
```

Threadpool metrics
```
tomcat_threads_busy{name="http-nio-8080",} 1.0
tomcat_threads_current{name="http-nio-8080",} 10.0
tomcat_threads_config_max{name="http-nio-8080",} 200.0
```

Servlet/Global metrics
```
tomcat_servlet_request_duration_seconds_count{name="default",} 0.0
tomcat_servlet_request_duration_seconds_sum{name="default",} 0.0
tomcat_servlet_request_duration_seconds_count{name="dispatcherServlet",} 7.0
tomcat_servlet_request_duration_seconds_sum{name="dispatcherServlet",} 0.182
tomcat_servlet_error_total{name="default",} 0.0
tomcat_servlet_error_total{name="dispatcherServlet",} 0.0
tomcat_servlet_request_max_seconds{name="default",} 0.0
tomcat_servlet_request_max_seconds{name="dispatcherServlet",} 0.144

tomcat_global_request_max_seconds{name="http-nio-8080",} 0.0
tomcat_global_request_duration_seconds_count{name="http-nio-8080",} 4.0
tomcat_global_request_duration_seconds_sum{name="http-nio-8080",} 7.317
tomcat_global_sent_bytes_total{name="http-nio-8080",} 0.0
tomcat_global_received_bytes_total{name="http-nio-8080",} 0.0
```

The `name` label is the `mbean` `name` property since it is possible to have multiple servlets, and possible to have multiple globals (I'm guessing by having multiple Tomcat instances embedded on separate ports in the same JVM)

I wasn't able to use the JMX Seraver in the unit tests and ended up mocking them out. I also don't love the extra thread that polls to register once it detects the JMX registration has occurred. I'm including it for now, until I can propose an improvement to Micrometer that will allow for a better hook.

FYI @jkschneider @mweirauch

### CODE REVIEW

**Comment by mweirauch** [comment]
From a plotting perspective this LGTM in case I got everything right.
2017-12-04-21-29-localhost

File: micrometer-core/src/main/java/io/micrometer/core/instrument/binder/tomcat/TomcatMetrics.java [hunk 1]
```diff
@@ -72,4 +101,145 @@ public void bindTo(MeterRegistry reg) {
...
+        //noinspection InfiniteLoopStatement
+        while (System.currentTimeMillis() < endWait) {
+            try {
+                if(!server.queryNames(new ObjectName("Tomcat:type=GlobalRequestProcessor,*"), null).isEmpty()){
+                    return;
+                }
+
+                Thread.sleep(500);
+            } catch (Exception e) {
+                e.printStackTrace();
```
Comment (jkschneider) [inline micrometer-core/src/main/java/io/micrometer/core/instrument/binder/tomcat/TomcatMetrics.java]: Nit for me to address.

Comment (checketts) [reply micrometer-core/src/main/java/io/micrometer/core/instrument/binder/tomcat/TomcatMetrics.java]: Would you like it to be a no-op or use SLF4J?

**Comment by jkschneider** [comment]
Merged with 3252d4a11abd3e31a06319f896788a10eff93691

**Pull request closed without merging**